app/src/main/java/amu/areeb/zhcet/utils/Utils.java

```diff
@@ -23,16 +23,16 @@
             "#FF6E40" /*Deep Orange*/};
 
     public static boolean isFacultyNumber(String fcNo) {
-        if (fcNo.length() == 9)
+        if (fcNo.length() == 8)
             return true;
-        else if (fcNo.length() != 8)
+        else if (fcNo.length() != 9)
             return false;
 
         String fYear = fcNo.substring(0, 2);
         String fBranch = fcNo.substring(2, 5);
         String fRNo = fcNo.substring(5);
 
-        return TextUtils.isDigitsOnly(fYear) && TextUtils.isDigitsOnly(fRNo) && fBranch.matches("^[ACEKLMP][EKR][B]$+") && Integer.parseInt(fYear) <= getSmallYear();
+        return TextUtils.isDigitsOnly(fYear) && TextUtils.isDigitsOnly(fRNo) && fBranch.matches("^[CEMP][ELKH][B]$+") && Integer.parseInt(fYear) <= getSmallYear();
```

Comment:
If we have this return statement do we really need those if else conditions on length? Can't we add that condition here @iamareebjamal

Reply:
If these things are unnecessary, just remove the conditions on faculty number completely. Allow anything to be used as faculty number

Reply:
These things are necessary. I will put the condition on faculty number length here which should be equal to 8

Reply:
But then I can't use the app as my faculty number is 14PEB049

Reply:
So what should I do? Are we okay with current changes?

Reply:
> But then I can't use the app as my faculty number is 14PEB049

You faculty number can't be found on the ctengg website.

Reply:
Yes, but it'll work in the app and API

Reply:
> So what should I do?

Remove all checks for faculty member or make it work for both kinds of faculty numbers

Reply:
@iamareebjamal We can do one thing to make it work for all kinds of validations. Just removing the validation through branch regex and keeping all other things the same.